include/seqan3/alphabet/mask/mask.hpp

```diff
@@ -0,0 +1,154 @@
+//
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+
+/*!\file
+ * \author Joshua Kim <joshua.kim AT fu-berlin.de>
+ * \brief Create a mask composition which can be applied with another alphabet.
+ */
+
+#pragma once
+
+#include <cassert>
+#include <seqan3/alphabet/concept_pre.hpp>
+
+namespace seqan3
+{
+/*!\brief Implementation of a masked alphabet to be used for cartesian compositions.
+ * \ingroup mask
+ * \implements seqan3::semi_alphabet_concept
+ *
+ * \details
+ * This alphabet is not usually used directly, but instead via seqan3::masked_composition.
+ * For more information see the \link mask Mask submodule \endlink.
+ */
+struct mask
+{
+    //!\brief The type of the alphabet when represented as a number (e.g. via to_rank()).
+    using rank_type = bool;
+
+    /*!\name Boolean values
+     * \brief Static member "booleans" that can be assigned to the alphabet or used in aggregate initialization.
+     * \details Similar to an Enum interface . *Don't worry about the `internal_type`.
+     */
+    //!\{
+    static const mask UNMASKED;
+    static const mask MASKED;
+    //!\}
+
+    //!\brief The size of the alphabet, i.e. the number of different values it can take.
+    static constexpr rank_type value_size{2};
```

Comment:
```cpp
using rank_type = bool;
static constexpr rank_type value_size{2};
```
The value of value_size will be 1, so in Debug mode the assertion `assert(c < value_size);` will fail for MASKED/1/true.